src/pages/index.js

```diff
@@ -2,76 +2,215 @@ import "./index.css";
 const cardAddButton = document.querySelector(".profile__add-button");
 const addModal = document.querySelector("#profile-add-card");
-
 const profileEditForm = profileEditModal.querySelector("#edit-modal-form");
 const nameInput = document.querySelector("#profile-title-input");
 const jobInput = document.querySelector("#profile-description-input");
 const cardAddForm = addModal.querySelector("#add-modal-form");
-const cardListEl = document.querySelector(".cards__list");
+const profilePicture = document.querySelector("#profile-change-button");
+const pictureChangeModal = document.querySelector("#profile-change-picture");
+const pictureChangeForm = pictureChangeModal.querySelector(
+  "#picture-modal-form"
+);
+const deleteFormButton = document.querySelector(".cards__delete-button");
 
 const cardTemplateSelector = "#card-template";
 
-const editCardModal = new PopupWithForm(
-  "#profile-edit-modal",
-  handleProfileEditSubmit
-);
-editCardModal.setEventListeners();
+const api = new Api({
+  baseUrl: "https://around-api.en.tripleten-services.com/v1",
+  authToken: "6a75e942-1cdf-4570-be93-f1316f67106a",
+});
 
-const previewImageModal = new PopupWithImage("#preview-image-modal");
-previewImageModal.setEventListeners();
-const userInfo = new UserInfo(".profile__title", ".profile__description");
-const itemSection = new Section({
-  items: initialCards,
-  renderer: renderCard,
-  cssSelector: ".cards__list",
+let itemSection;
+
+api.getCardList().then((items) => {
+  itemSection = new Section({
+    items,
+    renderer: renderCard,
+    cssSelector: ".cards__list",
+  });
+
+  itemSection.renderItems(items);
 });
-itemSection.renderItems();
 
-const addCardModal = new PopupWithForm(
-  "#profile-add-card",
-  handleCardAddSubmit
+const userInfo = new UserInfo(
+  ".profile__title",
+  ".profile__description",
+  ".profile__image"
 );
-addCardModal.setEventListeners();
 
-function handleImageClick(name, link) {
-  previewImageModal.open(name, link);
+function renderCard(card) {
+  const cardElement = createCard(card);
+  itemSection.addItem(cardElement);
 }
 
-function createCard(cardData) {
-  return new Card(cardData, cardTemplateSelector, handleImageClick).getView();
+api
+  .getUserInfo()
+  .then((userData) => {
+    userInfo.setUserInfo(userData);
+    // a = true
+  })
+  .catch((err) => {
+    console.error(err);
+  });
+
+const deleteConfirm = new PopupWithConfirmation("#card-delete-modal");
+const handleDeleteConfirm = (card) =>
+  api
+    .deleteCard(card._id)
+    .then(() => {
+      console.log(card);
+      card.remove();
+    })
+    .catch((err) => {
+      console.log(err);
+    });
+deleteConfirm.setEventListeners();
+
+function handleDeleteClick(card) {
+  // Set the submit action for the confirmation popup
+  console.log(card._id);
+  // why don't we open confirm modal first?
+  deleteConfirm.setSubmitAction(() => {
+    console.log(deleteConfirm);
+    handleDeleteConfirm(card);
+    deleteConfirm.close();
```

Comment:
* You need to close popups only in blocks `then` after a **successful response** from the server so that when there is an error it would be opened (the inputs will not be cleared and the user could send the data again to the server) and if you change the button text you could see the change as well